tests/test_reporters/test_html_server.py

```diff
@@ -89,3 +111,83 @@ def test_open_html_in_browser_watch(snapshot):
 
     finally:
         process.send_signal(signal.SIGINT)
+        process.wait()
+
+
+@pytest.fixture(scope="function")
+def temp_script_file_path(tmp_path) -> str:
+    script_path = os.path.normpath(
+        os.path.join(__file__, "../../fixtures/reporters/watch_integration.py")
+    )
+    with open(script_path, "r") as file:
+        lines = file.readlines()
+
+    file_path = os.path.join(tmp_path, "watch_integration.py")
+    with open(file_path, "w") as temp_file:
+        temp_file.writelines(lines)
+
+    return file_path
+
+
+def test_watch_update(temp_script_file_path, snapshot):
+    """Test the start_server_once function with watch=True using a fixed port.
+    Ensure the server changes the report contents after making code changes"""
+
+    process = subprocess.Popen([sys.executable, temp_script_file_path])
+
+    if not wait_for_server(5008):
+        process.send_signal(signal.SIGINT)
+        process.wait()
+        pytest.fail("Server did not start within the expected timeout")
+
+    try:
+        conn = HTTPConnection("127.0.0.1", 5008)
+        conn.request("GET", "/")
+        response = conn.getresponse()
+        assert response.status == 200
+
+        response_body = response.read().decode("utf-8")
+        cleaned_body_before = clean_response_body(response_body)
+        snapshot.assert_match(cleaned_body_before, "watch_html_server_snapshot.html")
+
+        with open(temp_script_file_path, "a") as py_file:
+            py_file.write("# This doesn't belong here!")
+
+        cleaned_body_after = updated_html_from_server(cleaned_body_before)
+        if cleaned_body_after is None:
+            pytest.fail("Server did not update HTML code served")
+        else:
+            snapshot.assert_match(cleaned_body_after, "watch_html_server_snapshot_updated.html")
+
+    finally:
+        process.send_signal(signal.SIGINT)
+        process.wait()
+
+
+def test_websocket_message(temp_script_file_path):
+    """Test that the "reload" message is sent to any open websocket connections
+    upon update to the files being watched
+    """
+    process = subprocess.Popen([sys.executable, temp_script_file_path])
+
+    try:
+        if not wait_for_server(5008):
+            pytest.fail("Server did not start within the expected timeout")
+
+        ws = websocket.create_connection("ws://localhost:5008/ws")
+        ws.settimeout(1)
+
+        with open(temp_script_file_path, "a") as py_file:
+            py_file.write("# trigger reload\n")
+
+        time.sleep(2)  # give the server time to send the websocket message
```

Comment:
Okay, I looked into this a little bit. I don't think you need to call `time.sleep`, you can set a larger timeout for `ws`, e.g. 10s.

Also I think you can just pass a `timeout=10` argument to `websocket.create_connection` rather than calling `ws.settimeout` immediately after creating it.